src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -632,104 +518,39 @@ private ODataBatchOperationRequestMessage CreateOperationRequestMessageImplement
                 this.payloadUriConverter.AddContentId(this.currentOperationContentId);
             }
 
-            this.InterceptException(() => uri = ODataBatchUtils.CreateOperationRequestUri(uri, this.rawOutputContext.MessageWriterSettings.BaseUri, this.payloadUriConverter));
+            this.InterceptException(() =>
+                uri = ODataBatchUtils.CreateOperationRequestUri(uri, this.outputContext.MessageWriterSettings.BaseUri, this.payloadUriConverter));
 
-            // create the new request operation
-            this.CurrentOperationRequestMessage = ODataBatchOperationRequestMessage.CreateWriteMessage(
-                this.rawOutputContext.OutputStream,
-                method,
-                uri,
-                /*operationListener*/ this,
-                this.payloadUriConverter,
-                this.container);
+            this.CurrentOperationRequestMessage = this.CreateOperationRequestMessageImplementation(method, uri, contentId, payloadUriOption);
 
-            if (this.changeSetBoundary != null)
+            if (this.isInsideSubBatch)
             {
                 this.RememberContentIdHeader(contentId);
             }

```

Comment:
I think we should do this regardless of whether or not we're in a changeset.  Legacy behavior was that you could only reference ids within the same changeset, but that has since been relaxed. Maybe make this:
      if (this.isInsideSubBatch || this.inputContext.MessageReaderSettings.MaxProtocolVersion > ODataVersion.V4)

Reply:
Yes, make sense. Added the OR clause for <=V4.

---
In reply to: [145851677](https://github.com/OData/odata.net/pull/960#discussion_r145851677) [](ancestors = 145851677)